### DISCUSSION TITLE
Battery level with additional load

### DISCUSSION THREAD

**Discussion by Rimbaldo** [rlogiacco/BatterySense#30] [Ideas]
My circuit is powered by a 7,4V battery. The battery is connected to my Leonardo VIN. When I turn on a 3W led, with it's driver circuit  powered directly by the battery, and enabled/disabled and controlled it's intensity by the Arduino, the voltage level in the battery drops. 

Do you have any functions that allow to declare something like a "load voltage drop" in the battery initialization options? Something like if a load is turned on (detected by an input pin,, or pwm different than 0), it then corrects the voltage readings? 
 
Regards, Rodrigo

**Comment by rlogiacco** [comment]
Hi @Rimbaldo ,
No, the library doesn’t provide anything like that, but you can change the library to compensate and normalize the readings on your needs...

**Comment by rlogiacco** [comment]
BTW, you might want to submit a PR with your changes and I might incorporate that feature in the library, if you like to contribute it

**Comment by Rimbaldo** [comment]
Ok, I'll try later then... Thanks!!

**Comment by rlogiacco** [comment]
On second thoughts, you have a couple of options other than changing the library to compensate: 
1. briefly turn off the load before invoking the `voltage` or `level` function, consequently removing the voltage drop from the equation
2. if the load needs to be always on, then incorporate the voltage drop in your battery max/min reference voltages, subtracting the voltage drop of the load from your battery max and min voltages
3. in case none of the above is fine, define two separate instances of the `Battery` class: you use the most appropriate depending on the status of the load, with one class using max and min values configured as no load is connected, the other configured accommodating the max and min voltages to the load voltage drop

As stated, I would welcome a contribution providing a flexible method for including a voltage drop, probably something in the order of a function to retrieve the expected min/max values

**Comment by rlogiacco** [comment]
The OP never replied
